### ISSUE TITLE
Type mismatch error when running runqlen.bt

### ISSUE THREAD

**Issue by jeromemarchand** [iovisor/bpftrace#196]
I received the following error when running runqlen.bt:

# bpftrace  /usr/local/share/bpftrace/tools/runqlen.bt 
Type mismatch for $len: trying to assign value of type 'integer'
	when variable already contains a value of type 'integer'

As it turned out the faulty line is this one:
$len = $len > 0 ? $len - 1 : 0;	// subtract currently runing task

It seems that the same variable can't be on both side of an assignment. I couldn't find anything about it in the reference guide, so I'm not sure if that's a language limitation or an implementation error. Anyway, the following patch fixes the problem:

```
diff --git a/tools/runqlen.bt b/tools/runqlen.bt
index 117f026..9d7f048 100644
--- a/tools/runqlen.bt
+++ b/tools/runqlen.bt
@@ -31,8 +31,8 @@ profile:hz:99
 {
        $task = (task_struct *)curtask;
        $my_q = (cfs_rq_partial *)$task->se.cfs_rq;
-       $len = $my_q->nr_running;
-       $len = $len > 0 ? $len - 1 : 0; // subtract currently runing task
+       $tmp = $my_q->nr_running;
+       $len = $tmp > 0 ? $tmp - 1 : 0; // subtract currently runing task
        @runqlen = lhist($len, 0, 100, 1);
 }
```

**Comment by brendangregg** [comment]
Thanks for debugging, this is a bpftrace bug -- that tool should work as is. What's your kernel and llvm version? I developed the tool on llvm 5, and my guess is that llvm has extra checks by the later versions.

**Comment by ajor** [comment]
This was a bug in the semantic analyser introduced after runqlat was written. Fixed in 4efac77775229def94b97c9ef804423baa497dc8
